### ISSUE TITLE
Gmod addons directory for GMA extractor removed all content without prompt or warning

### ISSUE THREAD

**Issue by Shadowysn** [WilliamVenner/gmpublisher#81]
EDIT: Heat of the moment had me unable to report properly. Sorry for the inconvenience.
Copy-past from my reply:
When I used 'browse' to set a custom location, other files and folders were not affected and the extracted files were properly put into a new folder.
Whereas using the 'Addons' option resulted in the extracted files being put directly inside the addons directory without making a new folder, and moving all other folders and files to the recycle bin.
(I have my steam installation in C: and my steam library in a different drive btw)

Original comment:
When I tried to use the GMA extractor, I thought the 'Existing files will be overridden' warning only applied to same-name files and the rest would be untouched, and expected the GMA contents to be put into a new, separate folder.

My collection of personal mod files which I stored in the addons folder as a Lua coder were removed by the extracted files which were put directly in the addons directory, and I tried to find my stuff back but accidentally cleared my Recycle bin out of habit, which had all of it inside. There was no indication they were going to be put into the Recycle bin.
Worst part is the extracted GMA files were properly put in a separate folder if I chose a custom location instead, and none of the files there were erased.

I'm honestly very angry with this.

**Comment by WilliamVenner** [comment]
I don't know what you want me to say or do. You were given warning files would be overwritten/lost, and you even have the choice to put removed files in the recycle bin or to permanently delete them. This is simply user error, you cannot be angry at the software for using it incorrectly when it is doing what it is programmed to do. Files are deleted because deleted files that are kept around can quite literally cause errors and bugs in addons and are totally unexpected to the end-user. There is nothing I can or will do about this.

**Comment by WilliamVenner** [comment]
Moving forward, don't blindly clear your recycle bin, pay attention when a program warns you about overwriting files, and try a data recovery tool. The files aren't really gone yet, not until the disk sectors are overwritten. Only clear your recycle bin if it's actually becoming a problem. I have found year-old files that I need in my recycle bin because I almost never clear it.

**Comment by Shadowysn** [comment]
Sorry for blasting off, I have a hard time collecting my thoughts and controlling myself in an emotional state.
I retrieved an older backup of my files. Recuva does a poor job recovering.
It was mostly user-error, yes, but it was also instigated by what looks to be a bug or at-least an inconsistency with the addons directory.

When I used 'browse' to set a custom location, other files and folders were not affected and the extracted files were properly put into a new folder.
Whereas using the 'Addons' option resulted in the extracted files being put directly inside the addons directory without making a new folder, and moving all other folders and files to the recycle bin.
